src/core/interp.c

```diff
@@ -696,7 +696,8 @@ void MVM_interp_run(MVMThreadContext *tc, void (*initial_invoke)(MVMThreadContex
             OP(abs_n):
                 {
                     MVMnum64 num = GET_REG(cur_op, 2).n64;
-                    if (num < 0) num = num * -1;
+                    // The 1.0/num logic checks for a negative zero
+                    if (num < 0 || num == 0 && 1.0/num < 0 ) num = num * -1;
```

Comment:
The division by zero niggled me enough to do some more googling, and I think the correct way to use this is to use the `copysignl` function from `math.h`. I think `num = copysign(num, 1.0)` should set the sign to be positive, for all values.